driver-core/src/main/com/mongodb/connection/CommandMessage.java

```diff
@@ -48,6 +49,8 @@
     private final SplittablePayload payload;
     private final FieldNameValidator payloadFieldNameValidator;
     private final boolean responseExpected;
+    private boolean messageHasBeenEncoded;
+    private boolean updatedResponseExpected;
```

Comment:
You can ditch this boolean and compute lazily in isResponseExpected().  The property name is confusing anyway, so it would aid clarity

Reply:
Done.